kaldb/src/test/java/com/slack/kaldb/metadata/replica/ReplicaMetadataSerializerTest.java

```diff
@@ -0,0 +1,47 @@
+package com.slack.kaldb.metadata.replica;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.junit.Assert.*;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import java.time.Instant;
+import org.junit.Test;
+
+public class ReplicaMetadataSerializerTest {
+  private final ReplicaMetadataSerializer serDe = new ReplicaMetadataSerializer();
+
+  @Test
+  public void testCacheSlotMetadataSerializer() throws InvalidProtocolBufferException {
+    String name = "name";
+    String snapshotId = "snapshotId";
+    long createdTimeUtc = Instant.now().toEpochMilli();
+
+    ReplicaMetadata replicaMetadata = new ReplicaMetadata(name, snapshotId, createdTimeUtc);
+
+    String serializedReplicaMetadata = serDe.toJsonStr(replicaMetadata);
+    assertThat(serializedReplicaMetadata).isNotEmpty();
+
+    ReplicaMetadata deserializedReplicaMetadata = serDe.fromJsonStr(serializedReplicaMetadata);
+    assertThat(deserializedReplicaMetadata).isEqualTo(replicaMetadata);
+
+    assertThat(deserializedReplicaMetadata.name).isEqualTo(name);
+    assertThat(deserializedReplicaMetadata.snapshotId).isEqualTo(snapshotId);
+    assertThat(deserializedReplicaMetadata.createdTimeUtc).isEqualTo(createdTimeUtc);
+  }
+
+  @Test
+  public void testInvalidSerializations() {
+    Throwable serializeNull = catchThrowable(() -> serDe.toJsonStr(null));
```

Comment:
Replace with assertThatIllegalArgumentException().isThrownBy()?

Reply:
This code has been moved to #161.

This is based off of the existing serialization tests from `SearchMetadataSerializerTest.testInvalidSerializations` and follows the same pattern established there.